indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -473,15 +500,28 @@ public Response getPendingTasks(@Context final HttpServletRequest req)
   @GET
   @Path("/runningTasks")
   @Produces(MediaType.APPLICATION_JSON)
-  public Response getRunningTasks(@Context final HttpServletRequest req)
+  public Response getRunningTasks(
+      @QueryParam("type") String taskType,
```

Comment:
so all the taskType business is added to support this use case ? why did we need this ?

Reply:
First, in this PR, it's needed for coordinators to find only running ```compact``` tasks. At each run, the coordinator computes the available compaction task slots by (total worker capacity - running ```compact``` tasks).

Second, I'm planning to refactor the code around taskStatus. We have several classes relating taskStatus, and meta information (like taskLocation, task creation time, etc) is scattered in those classes. The meta information is important for clients as well as Druid itself, but this scattered information makes difficult to have the same view in all places.